python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -111,14 +114,20 @@ def asset2(asset1):
     )
 
     all_resource_defs = dict(resource_defs)
-    for asset_def in assets:
-        for resource_key, resource_def in asset_def.resource_defs.items():
+    for assets_def in assets:
+        # Until we remove the ability to specify resource defs externally here, cannot check all asset defs for satisfying their resource requirements.
+        if assets_def.resource_defs:
+            get_and_validate_transitive_resource_dependencies(
+                assets_def.resource_defs, assets_def.required_resource_keys
+            )
+        for resource_key, resource_def in assets_def.resource_defs.items():
             if (
                 resource_key in all_resource_defs
                 and all_resource_defs[resource_key] != resource_def
             ):
+                asset_keys_str = ", ".join([key for key in assets_def.asset_keys])
                 raise DagsterInvalidDefinitionError(
-                    f"When attempting to build job, asset {asset_def.asset_key} had a conflicting version of the same resource key {resource_key}. Please resolve this conflict by giving different keys to each resource definition."
+                    f"When attempting to build job, asset keys {asset_keys_str} had a conflicting version of the same resource key {resource_key}. Please resolve this conflict by giving different keys to each resource definition."
```

Comment:
I think it's a little bit inaccurate to say that the "asset keys" have conflicting versions of the same resource key. It's the assets that have resources - not the keys. Separately, if I read that "asset1, asset2" had a conflicting version of the same resource key, I would think that I needed to fix the problem by using the same resource key for asset1 and for asset2, which I don't think is the case, right?

Reply:
That's correct - good call out. Will change the wording here.